classes/converter.php

```diff
@@ -568,7 +499,77 @@ public function define_client_check_sdk() {
      * @throws /coding_exception
      */
     public function define_client_check() {
-        $converter = new \fileconverter_librelambda\converter();
-        return $converter::are_requirements_met('CONNECTION');
+        global $OUTPUT;
+
+        // First check that we have the basic configuration settings set.
+        if (!$this->is_config_set()) {
+            return $OUTPUT->notification(get_string('settings:confignotset', 'fileconverter_librelambda'), 'notifyproblem');
+        }
+
+        $this->create_client();
+
+        $result = $this->check_requirements();
+        if ($result->success) {
+            return $OUTPUT->notification(get_string('settings:connectionsuccess', 'fileconverter_librelambda'), 'notifysuccess');
+        } else {
+            return $OUTPUT->notification($result->message, 'notifyproblem');
+        }
+    }
+
+    /**
+     * Reset client
+     */
+    public function remove_client() {
+        $this->client = null;
+    }
+
+    /**
+     * Whether the plugin is configured and other checks.
+     * @return object
+     */
+    public function check_requirements() {
+        $result = new \stdClass();
+        $result->success = true;
+        $result->message = get_string('settings:permissioncheckpassed', 'fileconverter_librelambda');
```

Comment:
It's a little bit confusing why it's `permissioncheckpassed`.
It should be either `settings:connectionsuccess` either an empty string.
As this `$result->message` is never used when `$result->success` is true (we return other lang strings) I'd set it to an empty string here.